api/provider/aws/generic_load_balancer.go

```diff
@@ -0,0 +1,51 @@
+package aws
+
+import (
+	"github.com/aws/aws-sdk-go/service/elb"
+	alb "github.com/aws/aws-sdk-go/service/elbv2"
+)
+
+func newGenericLoadBalancer(CLB *elb.LoadBalancerDescription, ALB *alb.LoadBalancer) *genericLoadBalancer {
+	if CLB != nil {
+		return &genericLoadBalancer{
+			CLB:   CLB,
+			isCLB: true,
+		}
+	}
+
+	if ALB != nil {
+		return &genericLoadBalancer{
+			ALB:   ALB,
+			isALB: true,
+		}
+	}
+
+	return nil
+}
+
+type genericLoadBalancer struct {
+	CLB   *elb.LoadBalancerDescription
+	ALB   *alb.LoadBalancer
+	isALB bool
+	isCLB bool
+}
+
+func (c genericLoadBalancer) Scheme() *string {
```

Comment:
why deal with pointers here? Seems to make life more difficult for the callers

Reply:
✅